### PR TITLE
R4R: Stub out other RPC endpoints

### PR DESCRIPTION
**Pull request by mslipper**
- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated all relevant documentation (`docs/`)
- [ ] Updated all relevant code comments
- [ ] Wrote tests
- [ ] Added entries in `PENDING.md` that include links to the relevant issue or PR that most accurately describes the change.
___________________________________
For Admin Use:
- [x] Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- [x] Reviewers Assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Review by jackzampolin — COMMENTED** [review]
This looks good to me! Maybe we need to add some documentation here (`/docs/web3`) that notes methods with fixed returns (mining, uncles, etc...) and links to the web3 spec we are implementing.

File: Gopkg.lock [hunk 1]
```diff
@@ -3,62 +3,55 @@
 
 [[projects]]
   branch = "master"
-  digest = "1:fcdf62d2d7e43c2565d6f8707ab4eae54dac702ed4bafb194b85139f0508929f"
```
Comment (alexanderbez) [inline Gopkg.lock]: I don't think we should be getting this diff. Can you update your dev tools?

File: server/rpc/eth_api.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/ethereum/go-ethereum/signer/core"
+	"math/big"
+)
+
+// PublicEthAPI is the web3_ prefixed set of APIs in the Web3 JSON-RPC spec.
+type PublicEthAPI struct{}
+
+// NewPublicEthPI creates an instance of the public ETH Web3 API.
+func NewPublicEthPI() *PublicEthAPI {
```
Comment (alexanderbez) [inline server/rpc/eth_api.go]: `NewPublicEthAPI`*

File: server/rpc/eth_api.go [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+import (
+	"github.com/cosmos/ethermint/version"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/ethereum/go-ethereum/signer/core"
+	"math/big"
+)
+
+// PublicEthAPI is the web3_ prefixed set of APIs in the Web3 JSON-RPC spec.
```
Comment (alexanderbez) [inline server/rpc/eth_api.go]: This is the same description as `PublicWeb3API` -- what's the difference?

File: server/rpc/eth_api.go [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+}
+
+// Syncing returns whether or not the current node is syncing with other peers. Returns false if not, or a struct
+// outlining the state of the sync if it is.
+func (e *PublicEthAPI) Syncing() interface{} {
+	return false
+}
+
+// Coinbase returns this node's coinbase address. Not used in Ethermint.
+func (e *PublicEthAPI) Coinbase() common.Address {
```
Comment (alexanderbez) [inline server/rpc/eth_api.go]: You can just use a named return variable here and simply `return`

**Review by alexanderbez — CHANGES_REQUESTED** [review]
Thanks for the contrib @mslipper -- I left some feedback 👍

File: server/rpc/eth_api.go (hunk 3 above)
Comment (mslipper) [reply server/rpc/eth_api.go]: I meant to make it the eth_ prefixed APIs. Fixing.

File: Gopkg.lock (hunk 1 above)
Comment (mslipper) [reply Gopkg.lock]: Tools updated.

**Comment by alexanderbez** [comment]
Awesome! Thanks @mslipper. Going forward, if there are other issues you can think of in regards to Ethereum RPC/API related work, lets create issues for them and triage them 👍

**Review by alexanderbez — APPROVED** [review]

**Pull request merged**